File: LeetCode/Medium/0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;
        int k = n;
        int FLAG = 100000;
        for (int i = n-1; i >= 2; i--) {
            if (nums[i] == nums[i-2]) {
                nums[i] = FLAG;
                k --;
            }
        }
        int x = 0;
        int t = 0;

        while (true) {
            if (x == k) {
                break;
            }
            if (nums[t] == FLAG) {
                t++;
                continue;
            }
            nums[x] = nums[t];
            x++;
            t++;
        }
        return k;
    }
}

File: LeetCode/Medium/0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.go
func removeDuplicates(nums []int) int {
    prev := nums[0]
    read := 0
    write := 0
    cnt := 0

    n := len(nums)

    for (read < n) {

        if (prev == nums[read]) {
            if (cnt <= 1) {
                nums[write] = nums[read]
                write += 1
                read += 1
            } else {
                read += 1
            }
            cnt += 1

        } else {
            prev = nums[read]
            nums[write] = nums[read]
            write += 1
            read += 1
            cnt = 1
        }
    }

    return write
}